crates/jit-icache-coherence/src/libc.rs

```diff
@@ -60,6 +60,38 @@ pub(crate) fn pipeline_flush_mt() -> Result<()> {
         // In any other case we got an actual error, so lets propagate that up
         e => e?,
     }
+    #[cfg(all(target_arch = "riscv64", target_os = "linux"))]
+    if num_cpus::get() == 1 {
+        use std::arch::asm;
+        unsafe {
+            asm!("fence.i");
+        }
+    } else {
+        match unsafe {
+            libc::syscall(
+                {
+                    // libc don't define the special syscall,So define it by My own.
+                    // https://github.com/torvalds/linux/search?q=__NR_arch_specific_syscall
+                    #[allow(non_snake_case)]
+                    let __NR_arch_specific_syscall = 244;
+                    __NR_arch_specific_syscall + 15
+                },
+                // Current the parameter is not used,But defined. https://github.com/torvalds/linux/blob/4d1044fcb996e8de9b9ab392f4a767890e45202d/arch/riscv/kernel/sys_riscv.c#L64-L66
+                0, // start
+                0, // end
+                {
+                    #[allow(non_snake_case)]
+                    let SYS_RISCV_FLUSH_ICACHE_LOCAL = 1;
+                    #[allow(non_snake_case)]
+                    let SYS_RISCV_FLUSH_ICACHE_ALL = SYS_RISCV_FLUSH_ICACHE_LOCAL;
+                    SYS_RISCV_FLUSH_ICACHE_ALL
```

Comment:
I'm also a bit confused as to why the kernel decide to alias both of these settings, but it seems as it was just to represent the state of things at [the time](https://github.com/torvalds/linux/commit/921ebd8f2c081b3cf6c3b29ef4103eef3ff26054). I think we should be OK just referencing `SYS_RISCV_FLUSH_ICACHE_ALL` which is the operation that we want but adding a comment on the current situation.